tests/ps_test.go

```diff
@@ -1,44 +1,167 @@
 package tests
 
 import (
+	"fmt"
+	"math/rand"
+	"os"
+	"regexp"
+	"sort"
+	"strconv"
+	"strings"
+	"sync"
+
 	. "github.com/onsi/ginkgo"
-	// . "github.com/onsi/gomega"
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gbytes"
+	. "github.com/onsi/gomega/gexec"
 )
 
-var _ = Describe("deis", func() {
+var procsRegexp = `(%s-v\d+-[\w-]+) up \(v\d+\)`
+
+// scrapeProcs returns the sorted process names for an app from the given output.
+// It matches the current "deis ps" output for a healthy container:
+//   earthy-vocalist-v2-cmd-1d73e up (v2)
+//   myapp-v16-web-bujlq up (v16)
+func scrapeProcs(app string, output []byte) []string {
+	re := regexp.MustCompile(fmt.Sprintf(procsRegexp, app))
+	found := re.FindAllSubmatch(output, -1)
+	procs := make([]string, len(found))
+	for i := range found {
+		procs[i] = string(found[i][1])
+	}
+	sort.Strings(procs)
+	return procs
+}
+
+var _ = Describe("Processes", func() {
+
 	Context("with a deployed app", func() {
 
-		XIt("can scale upward", func() {
-			// "deis ps:scale web=5 --app=%s"
-			// "deis ps:list --app=%s"
-			// curl app
-			// "deis ps:scale web=1 --app=%s"
-			// "deis ps:list --app=%s"
-			// curl app
-		})
+		appName := getRandAppName()
+		appURL := strings.Replace(url, "deis", appName, 1)
+		once := &sync.Once{}
 
-		XIt("can scale down to 0", func() {
-			// "deis ps:scale web=0 --app=%s"
-			// "deis ps:list --app=%s"
-			// curl app
-			// "deis ps:scale web=1--app=%s"
-			// "deis ps:list --app=%s"
-			// curl app
+		BeforeEach(func() {
+			// Set up the Processes test app only once and assume the suite will clean up.
+			once.Do(func() {
+				os.Chdir("example-go")
```

Comment:
`deployApp` in #57 is great. I have some comments I'll leave over there. When that gets merged this should definitely be refactored to use that.